### PR TITLE
feat(evals): human-labelled judge calibration set (#870)

### LINKED ISSUE #870: evals: build a human-labeled ground-truth calibration set for the judge
## Summary

Build a one-time, human-labeled gold-reference set for the `judge`-matcher tasks, so the automated LLM-as-judge can be measured and validated against ground truth.

## Motivation

Today the LLM-as-judge (`gemini-2.5-flash`) is trusted blind — there is no measurement of how often it agrees with a human. A small fixed calibration set fixes that and is the prerequisite for any judge-model decision.

Human judgment belongs here as the **gold reference**, not as a per-run judge. Putting a human in the loop of every sweep would destroy the harness's fast/repeatable property (judgments scale as judge-matcher-tasks x repeats x models x re-baselines), and humans have their own drift/fatigue reliability problem. One-time labeling is feasible precisely because the suite is small (~54 tasks; only the prose-heavy subset has `judge` matchers).

## What to do

- Once #869 lands, collect the judge-matcher `(request, response, criterion)` tuples from a representative sweep.
- Have a human label each YES/NO.
- Commit the labeled set as a versioned gold-reference artifact in the repo.

## Uses

- Measure judge accuracy (agreement rate with human labels).
- Compare candidate judge models head-to-head (see judge-model issue).
- Re-validate whenever the judge or the task criteria change.

## Dependencies

Depends on #869 (the harness must persist response text + per-judge-matcher criteria first). Feeds the judge-model decision issue.

### PR DESCRIPTION
**Pull request by allenhutchison**
## Summary

Fixes #870.

Builds the **gold-reference set** that lets us measure how often the LLM-as-judge (currently `gemini-2.5-flash`) agrees with a human. This is the prerequisite for the judge-model decision (#871) and judge-accuracy measurement going forward — without ground truth, the judge is trusted blind.

The PR contains both:
1. **Tooling** to extract calibration tuples from a sweep result.
2. **The populated, hand-labelled gold set** from a full 54-task × 3-repeat sweep on `gemini-3.1-flash-lite` (git sha `c2880ba`).

## Changes

- **`evals/lib/calibration.mjs`** — pure `extractCalibrationTuples()` that walks the post-#869 result schema and pulls every `judge`-matcher detail across every run of every task, attaching the per-task `userMessage` from the task lookup. One tuple per `(task, run, judge-matcher)`.
- **`evals/calibrate-extract.mjs`** + **`npm run eval:calibrate-extract`** — CLI wrapping the library. Defaults to the latest result; accepts `--from=<path>`, `--out=<path>`, `--force`. Refuses to overwrite an existing calibration file by default so labels can't be wiped by an accidental re-extract.
- **`evals/calibration/judge-calibration.json`** — the populated gold set: **90 tuples across the 30 judge-matcher tasks, each hand-labelled YES or NO**.
- **`test/evals/calibration.test.ts`** — 12 unit tests covering judge-only filtering, indexing across repeated runs, judge-error propagation, edge cases.
- **`evals/README.md`** — new "Judge calibration" section documenting the workflow.

## Calibration result

The headline number on the current judge (`gemini-2.5-flash`):

| Metric | Value |
| --- | --- |
| Tuples labelled | 90 / 90 |
| Human YES | 84 |
| Human NO | 6 |
| Agreement with automated judge | **83 / 90 (92%)** |
| Disagreements | **7** |

The 7 disagreements skew toward **false negatives** (6 of 7 are auto-NO / human-YES) and cluster in two themes:

- **Cosmetic-formatting brittleness** — same content marked NO in one run and YES in another when the only difference is backticks or a preamble (`skill-activate-apply::2`, `sort-recent-projects::1`).
- **Rubric-vs-response-content mismatch** — the judge wants the response to enumerate facts that the *prompt* didn't ask for (`rewrite-section-inplace` ×3); the task verifies facts via vault assertions, so this is partly a criterion-design issue.
- **Genuine miss** — `decision-brief::2` had all four required elements but the judge said NO.
- **Single human-stricter case** — `conflict-recency-rule::2` reached the right answer via the wrong primary rule (Authority instead of the criterion's required Recency).

This is the signal that feeds #871: a candidate-judge head-to-head should run against these 90 tuples and report agreement vs. ground truth. The same set will also catch judge regressions when criteria or models change.

## Testing

- `npm test` — 2926 passing (12 new calibration tests).
- `npm run build` — clean.
- `npm run format-check` — clean.
- **Live sweep + extract validated end-to-end:** `npm run eval` produced `evals/results/2026-05-22T22-39-07-456Z.json` (pass^3 100%, solve^3 77.8%, 162 runs, $0.17, no ERROR verdicts); `npm run eval:calibrate-extract` extracted exactly the expected 90 tuples across 30 tasks.
- **Labelling done interactively in-session**, one tuple at a time then batched 3-per-task, with the human reading prompt + criterion + response + automated verdict for each.

### Required

- [x] I have read and agree to the [Contributing Guidelines](../CONTRIBUTING.md)
- [x] I have read and agree to the [AI Policy](../AI_POLICY.md)
- [x] This PR is linked to an approved issue where the approach was discussed with a maintainer
- [x] All CI checks pass (`npm test`, `npm run build`, `npm run format-check`)
- [x] I have tested this change on Desktop (unit suite + live sweep + extract + labelling — see "Testing")
- [ ] I have verified this change does not break Mobile — N/A, the eval harness and calibration tooling are desktop-only Node tools
- [x] Documentation has been updated (if applicable)
- [x] I understand that I must address all review comments from CodeRabbit and maintainers, or this PR may be closed

### AI-Generated Code

- [x] This PR includes AI-generated or AI-assisted code
- [x] AI tool(s) used: Claude Code (tooling + extraction); labelling decisions made by the repo owner
- [x] I have reviewed and understand all AI-generated code in this PR

## Summary by CodeRabbit

* **New Features**
  * Added a judge calibration workflow to extract and record judge evaluation tuples for human labeling and validation.

* **Documentation**
  * Added a "Judge calibration" guide describing calibration extraction, tuple structure, human labeling conventions, overwrite protection, and the eval:calibrate-extract command.

* **Tests**
  * Added tests covering extraction logic, provenance fields, filtering, indexing, edge cases, and lookup fallbacks.

[Review Change Stack](https://app.coderabbit.ai/change-stack/allenhutchison/obsidian-gemini/pull/887?utm_source=github_walkthrough&utm_medium=github&utm_campaign=change_stack)

### CODE REVIEW

File: evals/README.md [hunk 1]
```diff
@@ -426,6 +426,47 @@ within its output ceiling. The `results/` directory is gitignored, so
...
+
+A human reads each tuple's `criteria` and `response` and sets `human_label` to `"YES"` or `"NO"`. The automated verdict is shown alongside but should not anchor the human — the whole point is independent ground truth. Tuples with a non-null `judge_error` (judge unavailable / API error) should generally be left at `null` since there is no automated verdict to compare against.
+
+The extractor **refuses to overwrite** an existing `judge-calibration.json` by default (a fresh extract would wipe the labels). Pass `--force` only when intentionally rebuilding from a new sweep.
+
+### Output location
+
+```
+evals/calibration/judge-calibration.json   ← committed to git once labelled
+```
```
[the remark this answers is not in the document]
Comment (allenhutchison) [reply evals/README.md]: Fixed in 0c4f32eb1c4658549e0285cacec18665d9391129 — added the `text` language tag.
